Домашнее задание с банкоматом. В классе ATM_DEMO пример его работы. В…

DmitryLapitsky opened the pull request:
… файле Readme описание задания и пояснения к решению

DmitryLapitsky committed:
Домашнее задание с банкоматом. В классе ATM_DEMO пример его работы. В файле Readme описание задания и пояснения к решению

kataus commented on hw06_solid/src/main/java/atm/ATM.java:
```diff
@@ -0,0 +1,112 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class ATM {
```
Интерфейс - полезная штука

kataus commented on hw06_solid/src/main/java/atm/Cash10.java:
```diff
@@ -0,0 +1,8 @@
+package atm;
+
+public class Cash10 extends Money {
```
Будет гораздо проще если для номинала использовать enum

DmitryLapitsky committed:
Добавлен интерфейс, определяющий поведение номинала валюты
Добавлен enum класс с возможными названиями классов и их соответствующим номиналом
В классе Money убран лишний метод в связи с введением enum

Поправлена ошибка в ATM - поле atm теперь не статичное и не собирает значения за предыдущие запуски

DmitryLapitsky committed:
добавлен интерфейс ATM

DmitryLapitsky committed:
убрал классы Cash10-5000, вместо них теперь enum

kataus commented on hw06_solid/src/main/java/atm/Money.java:
```diff
@@ -0,0 +1,70 @@
+
+    private int numberOfBills;
+
+    private double type;
```
Тут лучше было оставить enum, было бы удобнее. И нагляднее

DmitryLapitsky committed:
замена double на moneytypes
